homework/index.js

```diff
@@ -30,13 +30,108 @@
     return elem;
   }
 
+  function update(val, dBase) {
+    const container = document.getElementById('container');
+    const leftDiv = createAndAppend('div', container, { class: 'left-div whiteframe' });
+    const table = createAndAppend('table', leftDiv);
+    const tbody = createAndAppend('tbody', table);
+    const rep = dBase[val];
+
+    for (let ix = 0; ix < 4; ix++) {
+      const row = createAndAppend('tr', tbody);
+      const tdFirst = createAndAppend('td', row, { class: 'label' });
+      const tdSec = createAndAppend('td', row);
+      const updateDate = new Date(rep.updated_at).toLocaleString();
+      // eslint-disable-next-line default-case
+      switch (ix) {
+        case 0:
+          tdFirst.textContent = 'Repository :';
+          createAndAppend('a', tdSec, { text: rep.name, href: rep.html_url });
+          break;
+        case 1:
+          tdFirst.textContent = 'Description :';
+          tdSec.textContent = rep.description;
+          break;
+        case 2:
+          tdFirst.textContent = 'Forks :';
+          tdSec.textContent = rep.forks;
+          break;
+        case 3:
+          tdFirst.textContent = 'Updated :';
+          tdSec.textContent = updateDate;
+          break;
+      }
+    }
+
+    const rightDiv = createAndAppend('div', container, { class: 'right-div whiteframe' });
+    createAndAppend('p', rightDiv, {
+      text: 'Contributions',
+      class: 'contributor-header',
+    });
+    const ul = createAndAppend('ul', rightDiv, { class: 'contributor-list' });
+    const urlCol = rep.contributors_url;
+    fetchJSON(urlCol, (e, dt) => {
+      if (e) {
+        // Error message will be shown
+      } else {
+        dt.forEach(cont => {
+          const li = createAndAppend('li', ul, {
+            class: 'contributor-item',
+            tabindex: 0,
+            'aria-label': cont.login,
+          });
+
+          createAndAppend('img', li, {
+            src: cont.avatar_url,
+            height: 48,
+            class: 'contributor-avatar',
+            alt: cont.login,
+          });
+
+          const contData = createAndAppend('div', li, { class: 'contributor-data' });
+          createAndAppend('div', contData, { text: cont.login });
+          createAndAppend('div', contData, {
+            text: cont.contributions,
+            class: 'contributor-badge',
+          });
+        });
+      }
+    });
+  }
+
+  function initialize(rt, repos) {
```

Comment:
I had to examine other parts of your code to find out what you mean with `rt`. Avoid cryptic names like this. How much more effort does it take to type the full word `root`. VSCode will auto-complete it for you anyway.

Reply:
I used rt instead of root. Because i thought it will throw error if i use the same name in different functions. But i guess there will be no error. Did I understand right?

Reply:
The 'error' you may get is a style error (reported by ESLint). Your code will still run. But you only get this 'error' if reusing a name that is already available in a higher level scope, for instance:

```js
function foo() {
   const root = ...
   function baz(root) {
                 ^ESLint error : shadowed name
   }
}
```

But this is okay:

```js
function baz(root) {
   ...
}

function foo() {
  const root = ...
  baz(root);
}
```

Reply:
I understood. Thank you.